Russell - L242.Bld_Inc_Elas_scenarios

russellhz opened the pull request:
There is one thing that we might want to change: the script uses A42.demand.csv (which is a pretty bizarre file) but only takes its sector name of “buildings”. It doesn’t actually use the data (if you can even call it data). Maybe we should drop this file.

#291

bpbond commented on R/zchunk_L242.Bld_Inc_Elas_scenarios.R:
```diff
@@ -34,55 +44,185 @@ module_socioeconomics_L242.Bld_Inc_Elas_scenarios_DISABLED <- function(command,
+    # Linearly interpolate income elasticity for each level of per-capita GDP,
+    # using the assumption data
+    L242.IncomeElasticity_bld_GCAM3 <- L242.pcgdp_GCAM3_R_Y %>%
+      mutate(income.elasticity = approx(x = A42.inc_elas$pcgdp_90thousUSD, y = A42.inc_elas$inc_elas,
```
Why not use the existing `gcamdata::approx_fun` (still need to change that name!)?

russellhz replied:
I may be wrong in my understanding of approx_fun, but I don't think that would work here. This is taking the gdp and income elasticity from another file and using that relationship to estimate income elasticity for different gdp data. Basically creating a linear model to generate predictions. That doesn't look to me what approx_fun is capable of handling.

bpbond commented on R/zchunk_L242.Bld_Inc_Elas_scenarios.R:
```diff
@@ -34,55 +44,185 @@ module_socioeconomics_L242.Bld_Inc_Elas_scenarios_DISABLED <- function(command,
+    # Split by scenario and remove scenario column from each tibble
+
+    L242.pcgdp_thous90USD_Scen_R_Y <- L242.pcgdp_thous90USD_Scen_R_Y %>%
+      split(.$scenario) %>%
```
Huh. I totally didn't understand what was happening here. Now do I think.

bpbond commented on R/zchunk_L242.Bld_Inc_Elas_scenarios.R:
```diff
@@ -34,55 +44,185 @@ module_socioeconomics_L242.Bld_Inc_Elas_scenarios_DISABLED <- function(command,
+
+    L242.pcgdp_thous90USD_Scen_R_Y[["gSSP2"]] %>%
+      add_title("Building Income Elasticity: gSSP2") %>%
+      add_units("Unitless (% change in service demand / % change in income)") %>%
```
Given how much repetition there is, does it make sense to put some of this repeated stuff in lines 94-98 above?

russellhz replied:
Hadn't thought of that. I will add that in.

russellhz commented:
@bpbond This is one is under the socioeconomics team, not my normal energy team. Does that mean we should have different reviewers?

bpbond commented:
Sorry, yes. Please change. Thanks!

On May 2, 2017, at 8:19 AM, russellhz <notifications@github.com<mailto:notifications@github.com>> wrote:


@bpbond<https://github.com/bpbond> This is one is under the socioeconomics team, not my normal energy team. Does that mean we should have different reviewers?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/JGCRI/gcamdata/pull/326#issuecomment-298617011>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AB3adApT3RmyWzHj9wcEB4FE-MA8dE2Gks5r1xyugaJpZM4NNdSR>.

rplzzz commented on inst/extdata/socioeconomics/A42.inc_elas.csv:
```diff
@@ -0,0 +1 @@
45,0.15
50,0.1
55,0.05
60,0
```
Could somebody convert the line endings on this before final merge?

bpbond replied:
Done.

d3y419 commented on R/zchunk_L242.Bld_Inc_Elas_scenarios.R:
```diff
@@ -1,88 +1,181 @@
-             "L101.Pop_thous_GCAM3_R_Y",
-             "L102.gdp_mil90usd_GCAM3_R_Y",
-             "L102.pcgdp_thous90USD_Scen_R_Y"))
+             FILE = "L101.Pop_thous_GCAM3_R_Y",
```
tagging for GCAM 3 legacy. no immediate action needed.

d3y419 commented:
this is a rarely used piece of code that replicates MiniCAM structure. 
looks like the code does what it should. i approve this code. 
two potential broader issues:
1. per @rplzzz , we're getting rid of GCAM3 legacy. this one includes it. 
2. per @bpbond https://github.com/JGCRI/gcamdata/issues/321 , this one doesn't have the necessary level 2 IDstring header to create xml input.

rplzzz commented on R/zchunk_L242.Bld_Inc_Elas_scenarios.R:
```diff
@@ -1,88 +1,181 @@
+      add_legacy_name("L242.IncomeElasticity_bld_SSP5") ->
+      L242.IncomeElasticity_bld_SSP5
+
+    return_data(L242.IncomeElasticity_bld_GCAM3,
```
As Haewon pointed out, we are no longer building the GCAM3 scenarios, so this output should be removed, along with all of the code to build it.

russellhz commented:
I removed GCAM3 and switched one of the inputs from temp-data-inject. Looks like maybe we still need to wait to figure out how to deal with level 2 code?

bpbond commented:
@russellhz yeah that's probably true. I'll try to nail that down tomorrow.

d3y419 commented:
@rplzzz : approved?

bpbond commented:
Note - Level 2 ID strings not an issue here (another chunk will handle XML generation; see #358 ). I will add required files to this PR once we approve and merge #358 .
